docs: Add description for acceptable benchmark test results (#1458)
kyma-project/telemetry-manager: docs/contributor/benchmarks/README.md

|            0.107.0 |            19876            |            19719            |          0          |       184, 96        |     1, 1      |            11993             |            35504            |          0          |       133, 138       |   1.4, 1.4    |                   10081                    |             360             |         512         |      1734, 1722      |   0.5, 0.5    |                   18184                    |             922             |         510         |      1754, 1698      |    0.9, 1     |
|            0.108.1 |            20175            |            20014            |          0          |       100, 93        |     1, 1      |            12365             |            36603            |          0          |       135, 148       |   1.4, 1.4    |                   10758                    |             332             |         512         |      1722, 1681      |   0.5, 0.5    |                   19163                    |             891             |         510         |      1800, 1757      |     1, 1      |

</div>

> > **NOTE:** The expected throughput for the TracePipeline receiver is ~20K spans/sec for single pipeline scenario in other hand for multi pipeline scenario the expected throughput is ~13K spans/sec on receiver side and exported data should be 3 time of received spans. The backpressure scenarios should return the exporter queue size as ~512 for both single and multi pipeline scenarios and the TracePipeline instances should not crash.

Reviewer: ```suggestion
> **NOTE:** The expected throughput for the TracePipeline receiver on the receiver side is ~20K spans/sec for the single-pipeline scenario and ~13K spans/sec for the multi pipeline scenario. Exported data should be 3 times the received spans. The backpressure scenarios should return the exporter queue size as ~512, for both single- and multi-pipeline scenarios; and the TracePipeline instances should not crash.
```

> > Typical memory usage for TracePipeline instances is in average for single pod ~150MB and CPU usage is ~1.0 for non-backpressure scenarios, with backpressure scenarios the expected memory usage is significantly higher than non-backpressure scenarios on CPU side it should be less than non-backpressure scenarios since fewer data processed and exported compare to non-backpressure scenario. 

Reviewer: ```suggestion
> On average, memory usage for TracePipeline instances is ~150MB for a single Pod, and CPU usage is ~1.0 for non-backpressure scenarios.
> In backpressure scenarios, the behaviour is different: The expected CPU usage should be less than in non-backpressure scenarios, because less data is processed and exported compared to non-backpressure scenarios. The expected memory usage is significantly higher than in non-backpressure scenarios. 
```

|            0.107.0 |            19876            |            19719            |          0          |       184, 96        |     1, 1      |            11993             |            35504            |          0          |       133, 138       |   1.4, 1.4    |                   10081                    |             360             |         512         |      1734, 1722      |   0.5, 0.5    |                   18184                    |             922             |         510         |      1754, 1698      |    0.9, 1     |
|            0.108.1 |            20175            |            20014            |          0          |       100, 93        |     1, 1      |            12365             |            36603            |          0          |       135, 148       |   1.4, 1.4    |                   10758                    |             332             |         512         |      1722, 1681      |   0.5, 0.5    |                   19163                    |             891             |         510         |      1800, 1757      |     1, 1      |

</div>

> > **NOTE:** The expected throughput for the TracePipeline receiver on the receiver side is ~20K spans/sec for the single-pipeline scenario and ~13K spans/sec for the multi pipeline scenario. Exported data should be 3 times the received spans. The backpressure scenarios should return the exporter queue size as ~512, for both single- and multi-pipeline scenarios; and the TracePipeline instances should not crash.

Reviewer 2: I think an info like following is acceptable range would make this point more clear

</div>

> **NOTE:** The expected throughput for the TracePipeline receiver on the receiver side is ~20K spans/sec for the single-pipeline scenario and ~13K spans/sec for the multi pipeline scenario. Exported data should be 3 times the received spans. The backpressure scenarios should return the exporter queue size as ~512, for both single- and multi-pipeline scenarios; and the TracePipeline instances should not crash.
> On average, memory usage for TracePipeline instances is ~150MB for a single Pod, and CPU usage is ~1.0 for non-backpressure scenarios.

> > In backpressure scenarios, the behaviour is different: The expected CPU usage should be less than in non-backpressure scenarios, because less data is processed and exported compared to non-backpressure scenarios. The expected memory usage is significantly higher than in non-backpressure scenarios. 

Reviewer: ```suggestion
In backpressure scenarios, the behaviour is different: The expected CPU usage should be less than in non-backpressure scenarios, because less data is processed and exported compared to non-backpressure scenarios. The expected memory usage is significantly higher than in non-backpressure scenarios. 
```

</div>

> > **NOTE:** The expected throughput for the MetricPipeline receiver is ~4K metrics/sec for single pipeline scenario in other hand for multi pipeline scenario the expected throughput is ~3K metrics/sec on receiver side and exported data should be 3 time of received spans. The backpressure scenarios should return the exporter queue size between ~300 and 512 for both single and multi pipeline scenarios and the MetricPipeline instances should not crash.

Reviewer: ```suggestion
> **NOTE:** The expected throughput for the MetricPipeline receiver on the receiver side is ~4K metrics/sec for single-pipeline scenarios and ~3K metrics/sec for multi-pipeline scenarios. Exported data should be 3 times the received spans. The backpressure scenarios should return the exporter queue size between ~300 and 512 for both, single- and multi-pipeline scenarios; and the MetricPipeline instances should not crash.
```

> > Typical memory usage for MetricPipeline instances is in average for single pod ~150MB and CPU usage is ~1.5 for non-backpressure scenarios (multi pipeline scenario expect slightly higher) , with backpressure scenarios the expected memory usage is significantly higher than non-backpressure scenarios on CPU side it should be less than non-backpressure scenarios since fewer data processed and exported compare to non-backpressure scenario.

Reviewer: ```suggestion
> On average, memory usage for MetricPipeline instancesis ~150MB for a single Pod, and CPU usage is ~1.5 for non-backpressure scenarios (multi pipeline scenario expect slightly higher). In backpressure scenarios, the behaviour is different: The expected CPU usage should be less than in non-backpressure scenarios, because less data is processed and exported compared to non-backpressure scenarios. The expected memory usage is significantly higher than in non-backpressure scenarios.
```

</div>

> > **NOTE:** The expected throughput for the MetricPipeline receiver on the receiver side is ~4K metrics/sec for single-pipeline scenarios and ~3K metrics/sec for multi-pipeline scenarios. Exported data should be 3 times the received spans. The backpressure scenarios should return the exporter queue size between ~300 and 512 for both, single- and multi-pipeline scenarios; and the MetricPipeline instances should not crash.

Reviewer 2: Same like above a heading would make it more clear and visible

</div>

> > **NOTE:** The expected throughput for the MetricPipeline receiver on the receiver side is ~4K metrics/sec for single-pipeline scenarios and ~3K metrics/sec for multi-pipeline scenarios. Exported data should be 3 times the received spans. The backpressure scenarios should return the exporter queue size between ~300 and 512 for both, single- and multi-pipeline scenarios; and the MetricPipeline instances should not crash.

Reviewer: Same as above; let's turn this into normal paragraphs.

```suggestion
The expected throughput for the MetricPipeline receiver on the receiver side is ~4K metrics/sec for single-pipeline scenarios and ~3K metrics/sec for multi-pipeline scenarios. Exported data should be 3 times the received spans. The backpressure scenarios should return the exporter queue size between ~300 and 512 for both, single- and multi-pipeline scenarios; and the MetricPipeline instances should not crash.
```

> > On average, memory usage for MetricPipeline instancesis ~150MB for a single Pod, and CPU usage is ~1.5 for non-backpressure scenarios (multi pipeline scenario expect slightly higher). In backpressure scenarios, the behaviour is different: The expected CPU usage should be less than in non-backpressure scenarios, because less data is processed and exported compared to non-backpressure scenarios. The expected memory usage is significantly higher than in non-backpressure scenarios.

Reviewer: ```suggestion
On average, memory usage for MetricPipeline instances is ~150MB for a single Pod, and CPU usage is ~1.5 for non-backpressure scenarios (multi pipeline scenario expect slightly higher). In backpressure scenarios, the behaviour is different: The expected CPU usage should be less than in non-backpressure scenarios, because less data is processed and exported compared to non-backpressure scenarios. The expected memory usage is significantly higher than in non-backpressure scenarios.
```

> > Above-mentioned figures are minimum acceptable values but always compare the measured results with the previous execution from the table to check for performance degradations, e.g. higher CPU or memory usage, lower throughput, or higher queue size are indicators of performance degradation.

Reviewer: ```suggestion
> **NOTE:** The figures above are the minimum acceptable values. You should always compare the measured results with the previous execution from the table to check for performance degradations, typically indicated by higher CPU or memory usage, lower throughput, or higher queue size.
```

> > The test execution is automated, sometimes the test results may not be as expected, in such cases, the test should be re-executed to confirm the results.

Reviewer: ```suggestion
> The test execution is automated. If the test results aren't as expected, you should re-execute the test to confirm the results.
```

</div>

The expected throughput for the MetricPipeline receiver on the receiver side is ~4K metrics/sec for single-pipeline scenarios and ~3K metrics/sec for multi-pipeline scenarios. Exported data should be 3 times the received spans. The backpressure scenarios should return the exporter queue size between ~300 and 512 for both, single- and multi-pipeline scenarios; and the MetricPipeline instances should not crash.

> On average, memory usage for MetricPipeline instances is ~150MB for a single Pod, and CPU usage is ~1.5 for non-backpressure scenarios (multi pipeline scenario expect slightly higher). In backpressure scenarios, the behaviour is different: The expected CPU usage should be less than in non-backpressure scenarios, because less data is processed and exported compared to non-backpressure scenarios. The expected memory usage is significantly higher than in non-backpressure scenarios.

Reviewer: ```suggestion
On average, memory usage for MetricPipeline instances is ~150MB for a single Pod, and CPU usage is ~1.5 for non-backpressure scenarios (slightly higher for multi-pipeline scenarios). In backpressure scenarios, the behaviour is different: The expected CPU usage should be less than in non-backpressure scenarios, because less data is processed and exported compared to non-backpressure scenarios. The expected memory usage is significantly higher than in non-backpressure scenarios.
```

|            0.108.1 |             20219              |            20217             |          0          |       743, 751       |   0.2, 0.2    |                     20120                     |            20108             |          6          |       735, 780       |   0.2, 0.2    |
</div>

> > **NOTE:** The expected throughput for the MetricPipeline agent receiver is ~20K metrics/sec, expected memory usage is in average ~700MB and CPU usage is ~0.2 for each instance.

Reviewer: ```suggestion
> **NOTE:** The expected throughput for the MetricPipeline agent receiver is ~20K metrics/sec. Expected memory usage is on average ~700MB, and CPU usage is ~0.2 for each instance.
```

|            0.108.1 |             20219              |            20217             |          0          |       743, 751       |   0.2, 0.2    |                     20120                     |            20108             |          6          |       735, 780       |   0.2, 0.2    |
</div>

> > **NOTE:** The expected throughput for the MetricPipeline agent receiver is ~20K metrics/sec. Expected memory usage is on average ~700MB, and CPU usage is ~0.2 for each instance.

Reviewer: ```suggestion
The expected throughput for the MetricPipeline agent receiver is ~20K metrics/sec. Expected memory usage is on average ~700MB, and CPU usage is ~0.2 for each instance.
```

> > Above-mentioned figures are minimum acceptable values but always compare the measured results with the previous execution from the table to check for performance degradations, e.g. higher CPU or memory usage, lower throughput, or higher queue size are indicators of performance degradation.

Reviewer: ```suggestion
> **NOTE:** The figures above are the minimum acceptable values. You should always compare the measured results with the previous execution from the table to check for performance degradations, typically indicated by higher CPU or memory usage, lower throughput, or higher queue size.
```

> > The test execution is automated, sometimes the test results may not be as expected, in such cases, the test should be re-executed to confirm the results.

Reviewer: ```suggestion
> The test execution is automated. If the test results aren't as expected, you should re-execute the test to confirm the results.
```
